infra/utils.py

```diff
@@ -0,0 +1,82 @@
+from enum import Enum
+
+from google.cloud import run_v2
+
+PRODUCTION_INITIAL_TRAFFIC_PERCENT = 5
+
+
+class ProductionColor(Enum):
+    GREEN = "green"
+    BLUE = "blue"
+
+
+def get_latest_production_revision(
+    project_id: str, region: str, service_name: str
+) -> tuple[ProductionColor, str]:
+    revclient = run_v2.RevisionsClient()
+    service_name = "production-consent-api"
+    parent = f"projects/{project_id}/locations/{region}/services/{service_name}"
+
+    revisions = revclient.list_revisions(parent=parent)
+
+    sorted_revs = sorted(revisions.revisions, key=lambda r: r.create_time, reverse=True)
+
+    latest_rev = sorted_revs[0]
+    try:
+        col_annotation = latest_rev.annotations["production-color"]
```

Comment:
Question, non-blocking: I'm assuming for the first Blue/Green deployment, we'll manually add an annotation to the current revision?

Reply:
Of course, the screenshots are in the PR 🤦  Nice one!